src/app/components/Staking/FinalityProviders/FinalityProviderSearch.tsx

```diff
@@ -3,10 +3,12 @@ import React, { useEffect, useState } from "react";
 import { FiSearch } from "react-icons/fi";
 
 interface FinalityProviderSearchProps {
+  initialSearchValue: string | null;
   onSearch: (searchTerm: string) => void;
 }
 
 export const FinalityProviderSearch: React.FC<FinalityProviderSearchProps> = ({
+  initialSearchValue,
   onSearch,
 }) => {
   const [searchTerm, setSearchTerm] = useState("");
```

Comment:
I think this is enough. I don't think we need listen query params updates

```suggestion
  const [searchTerm, setSearchTerm] = useState(initialSearchValue);
```